java/Dockerfile

```diff
@@ -15,16 +15,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
-FROM maven
+FROM maven:3.6-jdk-8
 
-ADD header /arrow/
-ADD format /arrow/format
-ADD integration/data /arrow/integration/data
-ADD java /arrow/java
```

Comment:
Every source change invalidates the docker cache of this layer, so mvn needs to download the dependencies on each source edit, which is not desirable. A solution would be to recursively add each pom.xml files which is not supported by docker.